unit_test/test_spdm_responder/set_certificate_rsp.c

```diff
@@ -27,6 +27,14 @@ void libspdm_test_responder_set_cetificate_rsp_case1(void **state)
     size_t cert_chain_size;
     spdm_set_certificate_request_t *m_libspdm_set_certificate_request;
 
+#if defined(_MSC_VER) || (defined(__clang__) && (defined (LIBSPDM_CPU_AARCH64) || \

```

Comment:
why we need this?

Reply:
Because of the PR: https://github.com/DMTF/libspdm/commit/00885640d5dec19fcaa0051a3e6add56ccf363ee, there are two ways(fwrite or write) to write NVM. 
And when use `libspdm_read_responder_public_certificate_chain_per_slot` get the cert, it use fread to get cert. 
So add the MACRO to check for the `fwrite` way.

```
bool libspdm_write_certificate_to_nvm(uint8_t slot_id, const void * cert_chain,
                                      size_t cert_chain_size)
{
#if defined(_MSC_VER) || (defined(__clang__) && (defined (LIBSPDM_CPU_AARCH64) || \
    defined(LIBSPDM_CPU_ARM)))
    FILE *fp_out;
#else
    uint64_t fp_out;
#endif

    char file_name[] = {'s','l','o','t','_','i','d','_','0','\0'};

    file_name[strlen(file_name) - 1] = (char)(slot_id+'0');

#if defined(_MSC_VER) || (defined(__clang__) && (defined (LIBSPDM_CPU_AARCH64) || \
    defined(LIBSPDM_CPU_ARM)))
    if ((fp_out = fopen(file_name, "w+b")) == NULL) {
        printf("Unable to open file %s\n", file_name);
        return false;
    }

    if ((fwrite(cert_chain, 1, cert_chain_size, fp_out)) != cert_chain_size) {
        printf("Write output file error %s\n", file_name);
        fclose(fp_out);
        return false;
    }

    fclose(fp_out);
#else
    if ((fp_out = open(file_name, O_WRONLY | O_CREAT, S_IRWXU)) == -1) {
        printf("Unable to open file %s\n", file_name);
        return false;
    }

    if ((write(fp_out, cert_chain, cert_chain_size)) != cert_chain_size)
    {
        printf("Write output file error %s\n", file_name);
        close(fp_out);
        return false;
    }

    close(fp_out);
#endif

    return true;
}
```

Reply:
I dont see `else` branch.

How does that work in GCC?

Reply:
Thanks. I have changed the code. And the CI has passed.